front/app/containers/IdeasShow/index.tsx

```diff
@@ -183,20 +189,29 @@ export const IdeasShow = ({
   const { data: ideaImages } = useIdeaImages(ideaId);
 
   const [newIdeaId, setNewIdeaId] = useState<string | null>(null);
+  const [goBack, setGoBack] = useState(false);
   const [translateButtonIsClicked, setTranslateButtonIsClicked] =
     useState<boolean>(false);
-  const [queryParams] = useSearchParams();
-  const ideaIdParameter = queryParams.get('new_idea_id');
+  const [searchParams] = useSearchParams();
+  const ideaIdParameter = searchParams.get('new_idea_id');
+  const goBackParameter = searchParams.get('go_back');
   const timeout = useRef<NodeJS.Timeout>();
 
   useEffect(() => {
+    if (ideaIdParameter === null && goBackParameter === null) return;
+
     if (isString(ideaIdParameter)) {
       timeout.current = setTimeout(() => {
         setNewIdeaId(ideaIdParameter);
       }, 1500);
-      clHistory.replace(window.location.pathname);
     }
-  }, [ideaIdParameter]);
+
+    if (isString(goBackParameter)) {
```

Comment:
I don't fully understand this. If you have it in the parameters, why do you need it in the local component state, as well?

Reply:
We basically detect it, store it locally, and then remove it from the parameters. It's a mechanism we use in quite a few places. In this case it kind of makes sense, because if we pass `go_back` it makes sure that when you click the 'go back button' it uses the browser's native 'go back' functionality (which is necessary to return to the same scroll position). However when you share the URL to the project with someone else you don't want the button to behave like that, so the parameter shouldn't be there